src/main/java/seedu/address/model/student/Student.java

```diff
@@ -291,4 +293,35 @@ public String studentEssentialPrinter() {
                 .append(getAcademicYear());
         return builder.toString();
     }
+
+    /**
+     * Returns whether a student is able to attend a training at a particular DateTime
+     */
+    public boolean isAvailableAtDateTime(LocalDateTime dateTime) {
+        DayOfWeek day = dateTime.getDayOfWeek();
+
+        LocalTime queryTime = LocalTime.from(dateTime);
+        LocalTime studentTime;
+
+        switch (day) {
+        case MONDAY:
+            studentTime = mondayDismissal.dismissalTime;
```

Comment:
might be better to use getters to get the dismissalTime instead of accessing the variable directly